### ISSUE TITLE
[3.0.2.1b] password field not large enough (database)

### ISSUE THREAD

**Issue by ChrisLancs** [opencart/opencart#6207]
table users: After a password is rehashed and submitted to the database it is truncated to 40 characters preventing admin login there after, Field length needs increasing.

customer table is the same so likely the result will be the same for customer.

Steps to reproduce: change admin password (I didnt get this before the password change but then i also did not log out before changing the password after installing the system)

I fixed my login by changing the length of the password field in users table to 100, probably excessive but it clarified that the issue is just the length of the field.

**Comment by danielkerr** [comment]
all passwords have been changed to 255. check the master next time.

**Comment by ChrisLancs** [comment]
If you mean master that is version 3.0.2.1b which I used to create a fresh install, I think its fair to say that the SQL is not creating password fields of 255 length but 40.

Unless you have patched in the last two days and not incremented the version number. In which case, you can forgive me for not checking the masters code when the version number matches that which i installed.

**Comment by danielkerr** [comment]
where is 3.0.2.1b? b is the cloud version only.

https://github.com/opencart/opencart/releases

**Comment by danielkerr** [comment]
also github is not for old version. its the development of the next release. glad u fixed ur issue but we are always working for the master branch.

**Comment by ChrisLancs** [comment]
Yes i was looking to help on the development, I downloaded and installed the master from github onto a sub domain so its not my live site but a dev version.

Opening index.php in the site root shows the pre-mentioned version defined which still matches the current version here in the master repo.

The exact method of installation I followed was to fork your master repo.
Import my fork into netbeans
export from netbeans to my test sub-domain
Complete install.

I think the admin login after install completed the rehash but as I had not logged out to see it, it was only after i changed the admin password and attempted to log in that i found i couldn't. I found that changing the password to an md5 password no salt let me login once but not again. Hence i went digging and found the rehash.

After manually hashing the password and updating via phpmyadmin I noticed that the warning of being truncated to 40 characters was shown.

Anyway if its fixed its fixed, but two days ago at least the password fields with a length of 40 characters were being created.
